### ISSUE TITLE
<Tab> rendering <Button> component instead of native button

### ISSUE THREAD

**Issue by barteksekula** [material-components/material-components-web-react#920]
At the moment 'button' is hardcoded in the source code:
https://github.com/material-components/material-components-web-react/blob/master/packages/tab/index.tsx#L206
Would it be a good idea to support regular anchors?

The use case is to be able to render regular navigation links instead of a list of buttons with onClicks and navigation to a different url in javascript.

It would be enough if we just used the Button component instead of native html <button>
Then something like this would be possible:
```js
<Tab
  key={index}
  isFadingIndicator={isFadingIndicator}
  indicatorContent={indicatorContent}
  href="http://google.com"
  onClick={(event: React.MouseEvent<any>) => {
    event.preventDefault();
  }}
>
  <span className='mdc-tab__text-label'>Tab {num + 1}</span>
</Tab>
```
as the Button component is conditionally rendering button or a based on whether the href property is falsy or not:
https://github.com/material-components/material-components-web-react/blob/master/packages/button/index.tsx#L77

Proposed solution:
https://github.com/material-components/material-components-web-react/compare/master...barteksekula:master

**Comment by moog16** [comment]
So we'd be using the Button component, but not the styles? Could we instead just use something like `props.tag`? See https://github.com/material-components/material-components-web-react/blob/master/packages/list/ListItem.tsx#L207

**Comment by barteksekula** [comment]
Sure, that's also an option. I just thought it would be better to just utilize the existing Button component, especially that Tab's props are based on HTMLButtonElement, if we expose Tag, even with default prop set to 'button' users will still be able to set it to an element that is not intended to be used inside a Tab Bar.

I just took a quick look how other material implementation does that and they all seem to render some kind of Button/ButtonBase component as the first child of Tab component.

Would you be willing to accept a contribution with that change? Just tell me which one do you prefer and I will do it (either Button component or Tag prop).

Thanks!

**Comment by barteksekula** [comment]
two alternatives:
```js
export interface TabProps extends React.HTMLProps<HTMLButtonElement | HTMLAnchorElement> {
  active?: boolean;
  focusOnActivate?: boolean;
  isFadingIndicator?: boolean;
  indicatorContent?: React.ReactNode;
  minWidth?: boolean;
  isMinWidthIndicator?: boolean;
  stacked?: boolean;
  tag?: "button" | "a";
  previousIndicatorClientRect?: ClientRect;
  onInteraction?: () => void;
}
```

or just
```js
export interface TabProps extends React.HTMLProps<HTMLElement> {
  active?: boolean;
  focusOnActivate?: boolean;
  isFadingIndicator?: boolean;
  indicatorContent?: React.ReactNode;
  minWidth?: boolean;
  isMinWidthIndicator?: boolean;
  stacked?: boolean;
  tag?: string;
  previousIndicatorClientRect?: ClientRect;
  onInteraction?: () => void;
}
```

both with 
```js
defaultProps = {
  ...
  tag: 'button',
  ...
}
```

**Comment by moog16** [comment]
@barteksekula yeah that makes sense there is a button base, however it was not designed to be used within the tabs component. That might be a good update for MDC Web (vanilla version).

The React Button also has logic specific to the button that the tab will never use, which adds unnecessary cruft to the tab. Can we go with `Tag`?

Also `export interface TabProps extends React.HTMLProps<HTMLButtonElement | HTMLAnchorElement> {` makes more sense to me. This way users will not use an unintended element.

**Comment by moog16** [comment]
Closing this since we merged #928
